### PR TITLE
Sort tree items locally on rename

### LINKED ISSUE #5489: Object rename is unreliable
#### Summary
Renaming of objects works unreliably right now.

#### Expected vs Current Behavior
Renaming an object should work immediately or should inform the user of any errors.

#### Steps to Reproduce
1. Create a folder
2. Rename the folder by clicking on the folder's name in the browse bar
3. Observe that the folder's name reverts to the original name almost immediately
4. This issue is not always reproducible, but repeating steps 2 and 3 should eventually trigger it.

#### Environment
* Open MCT Version: 1c8784fec5ac2bcfb821b539a38d904e3c1cc926
* Deployment Type: VIPER testathon

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by ozyx**
Closes #5489 #5533 #5398

### Describe your changes:

Reorders the tree items locally on object rename. Previously, the tree was re-retrieving object compositions even though they had not changed, which caused stale objects to repopulate-- most noticeably while using CouchDB.

Also implements `aria-role: tree` for the mct-tree, and updates/adds some e2e `appActions` to make interacting with the tree a bit easier.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
While you're in there, can you add a component visual test for the tree in these various forms?

**Comment by ozyx** [comment]
> While you're in there, can you add a component visual test for the tree in these various forms?
> 
> How does the other internal tree behave inside of a crate or edit modal?

Yes! Let me give that a shot.

Good question re: the tree within create/edit modals. The appActions I've added are only intended for use with the main tree. The same tree component is used throughout, so the accessibility role is still applied to those other trees. The action is the same, so maybe we can extract the common actions to a private method and expose two different appActions, one for the main tree, and the other for modal trees?

**Comment by unlikelyzero** [comment]
> The action is the same, so maybe we can extract the common actions to a private method and expose two different appActions, one for the main tree, and the other for modal trees?

I would just make a test for the modal tree within this file and keep associated functions private to this file until we find a need to use them. Ideally, none of the tests outside of this suite would need be "interested" in that appAction, i would think

**Comment by ozyx** [comment]
> I would just make a test for the modal tree within this file and keep associated functions private to this file until we find a need to use them. Ideally, none of the tests outside of this suite would need be "interested" in that appAction, i would think

Good call, agree we should avoid overpopulating the appActions

File: e2e/appActions.js [hunk 1]
```diff
@@ -129,7 +124,7 @@ async function openObjectTreeContextMenu(page, myItemsFolderName, domainObjectNa
 async function getFocusedObjectUuid(page) {
     const UUIDv4Regexp = /[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}/gi;
     const focusedObjectUuid = await page.evaluate((regexp) => {
-        return window.location.href.match(regexp).at(-1);
+        return window.location.href.split('?')[0].match(regexp).at(-1);
```
Comment (scottbell) [inline e2e/appActions.js:127]: should we use a [regular URL object](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) here instead?

File: src/ui/layout/mct-tree.vue [hunk 2]
```diff
@@ -610,6 +615,44 @@ export default {
...
+            if (!this.isSortable(parentObjectPath)) {
+                return;
+            }
+
+            // Sort the renamed object and its siblings (direct descendants of the parent)
+            const directDescendants = this.getChildrenInTreeFor(parentItem, false);
+            directDescendants.sort(this.sortNameAscending);
+
+            // Take a copy of the sorted descendants array
+            const temp = directDescendants.slice();
```
Comment (scottbell) [inline src/ui/layout/mct-tree.vue:632]: can we use a better variable name please? 😀

**Review by scottbell — CHANGES_REQUESTED** [review]
Minor changes

File: e2e/appActions.js (hunk 1 above)
Comment (scottbell) [reply e2e/appActions.js:127]: We should not! The query parameters in the OpenMCT URLs are not really query parameters, even though they look like them.

Comment (scottbell) [reply e2e/appActions.js:127]: As they're all put after the `#`

**Review by scottbell — APPROVED** [review]
Strong work!

**Pull request merged**